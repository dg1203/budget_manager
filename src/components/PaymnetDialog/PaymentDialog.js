import React, { useState } from "react";
import {
  Button,
  Dialog,
  DialogTitle,
  Slide,
  TextField
} from "@material-ui/core";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import PropTypes from "prop-types";
import { useDispatch } from "react-redux";
import FormControl from "@material-ui/core/FormControl";
import { removeResource } from "../../actions/stateActions";

const Transition = React.forwardRef(function Transition(props, ref) {
  return <Slide direction="up" ref={ref} {...props} />;
});

const PaymentDialog = ({ currency, amount, jarId, open, openDialog }) => {
  const dispatch = useDispatch();
  const [paymentAmount, setPaymentAmount] = useState(0);
  const onSave = () => {
    dispatch(
      removeResource({
        currency,
        jarId,
        amount: paymentAmount
      })
    );
    setPaymentAmount(0);
    openDialog(false);
  };
  return (
    open && (
      <div>
        <Dialog
          open={open}
          TransitionComponent={Transition}
          keepMounted
          onClose={() => openDialog(false)}
          aria-labelledby="alert-dialog-slide-title"
          aria-describedby="alert-dialog-slide-description"
        >
          <DialogTitle id="alert-dialog-slide-title">
            Wypłata środków {currency}
          </DialogTitle>
          <DialogContent>
            <FormControl margin="normal" fullWidth variant="outlined">
              <TextField
                value={paymentAmount}
                onChange={event =>
                  setPaymentAmount(
                    parseFloat(event.target.value) > 0 &&
                      parseFloat(event.target.value) <= parseFloat(amount)
                      ? event.target.value
                      : paymentAmount
                  )
                }
                label="Kwota"
                variant="outlined"
                type="number"
                required
              />
            </FormControl>
          </DialogContent>
          <DialogActions>
            <Button onClick={onSave} color="primary">
              Zapisz
            </Button>
            <Button onClick={() => openDialog(false)} color="primary">
              Anuluj
            </Button>
          </DialogActions>
        </Dialog>
      </div>
    )
  );
};

PaymentDialog.propTypes = {
  open: PropTypes.bool.isRequired,
  openDialog: PropTypes.func.isRequired,
  jarId: PropTypes.string.isRequired,
  currency: PropTypes.string.isRequired,
  amount: PropTypes.string.isRequired
};

export default PaymentDialog;
