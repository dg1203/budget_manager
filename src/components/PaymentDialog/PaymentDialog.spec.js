import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { Provider } from "react-redux";
import PaymentDialog from "./PaymentDialog";
import store from "../../store";

const defaultProps = {
  currency: "PLN",
  amount: 200,
  jarId: "hdywekr",
  open: true,
  openDialog: jest.fn()
};

const renderComponent = () =>
  render(
    <Provider store={store}>
      <PaymentDialog {...defaultProps} />
    </Provider>
  );

describe("PaymentDialog component", () => {
  it("renders the component", () => {
    const container = renderComponent();
    expect(container).toMatchSnapshot();
  });
  it("test close method dialog", () => {
    const { getByTestId } = renderComponent();
    fireEvent.click(getByTestId("close-button"));
    expect(defaultProps.openDialog).toBeCalledWith(false);
  });
  it("test on save method", () => {
    const dispatchSpy = jest.spyOn(store, "dispatch");
    const { getByTestId } = renderComponent();
    fireEvent.click(getByTestId("save-button"));
    expect(dispatchSpy).toHaveBeenCalledWith({
      type: "REMOVE_RESOURCES",
      payload: {
        currency: defaultProps.currency,
        jarId: defaultProps.jarId,
        amount: 0
      }
    });
    expect(defaultProps.openDialog).toBeCalledWith(false);
  });
});
