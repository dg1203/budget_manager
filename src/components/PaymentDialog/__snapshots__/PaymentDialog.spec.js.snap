// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentDialog component renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="padding-right: 0px; overflow: hidden;"
  >
    <div
      aria-hidden="true"
    >
      <div
        data-testid="payment-dialog"
      />
    </div>
    <div
      class="MuiDialog-root"
      data-testid="dialog"
      role="presentation"
      style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
    >
      <div
        aria-hidden="true"
        class="MuiBackdrop-root"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      />
      <div
        data-test="sentinelStart"
        tabindex="0"
      />
      <div
        class="MuiDialog-container MuiDialog-scrollPaper"
        role="none presentation"
        style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
        tabindex="-1"
      >
        <div
          aria-describedby="alert-dialog-slide-description"
          aria-labelledby="alert-dialog-slide-title"
          class="MuiPaper-root MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiPaper-elevation24 MuiPaper-rounded"
          role="dialog"
        >
          <div
            class="MuiDialogTitle-root"
            id="alert-dialog-slide-title"
          >
            <h2
              class="MuiTypography-root MuiTypography-h6"
            >
              Wypłata środków 
              PLN
            </h2>
          </div>
          <div
            class="MuiDialogContent-root"
          >
            <div
              class="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth"
            >
              <div
                class="MuiFormControl-root MuiTextField-root"
              >
                <label
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled Mui-required Mui-required"
                  data-shrink="true"
                >
                  Kwota
                  <span
                    aria-hidden="true"
                    class="MuiFormLabel-asterisk MuiInputLabel-asterisk"
                  >
                     
                    *
                  </span>
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    required=""
                    type="number"
                    value="0"
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="PrivateNotchedOutline-legendLabelled-3 PrivateNotchedOutline-legendNotched-4"
                    >
                      <span>
                        Kwota
                         *
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
          </div>
          <div
            class="MuiDialogActions-root MuiDialogActions-spacing"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
              data-testid="save-button"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Zapisz
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
              data-testid="close-button"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Anuluj
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        data-test="sentinelEnd"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
  >
    <div
      data-testid="payment-dialog"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
